src/dmd/expressionsem.d

```diff
@@ -3013,13 +3016,19 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 // overload of opCall, therefore it's a call
                 if (exp.e1.op != TOK.type)
                 {
-                    if (sd.aliasthis && exp.e1.type != exp.att1)
+                    if (!exp.aliasthislock) //we don't need the recursion. On recursion done in iterateAliasThis
```

Comment:
"On recursion done in iterateAliasThis" -- I can't understand this sentence. Is there a typo somwhere?

Reply:
fixed